### DISCUSSION TITLE
leading zeros with handlebars

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#2865] [Help] [answered]
Is there a way to produce a defined number of leading zeros with the handlebars helpers?

**Comment by tobehn** [comment]
Ok, i realized this is not the easiest with handlebars as in my case i have a number in a cell (number circle) i want to increase by 1 on a button click. 
And then i want to check if it is less than 10 or, between 10 and 100 or greater than 100 in order to subsequently stringify it and prepend "000" , "00" or "0". By this i'd generate customer numbers, part numbers and any autogenerated numbers.

@mjashanks I guess its the perfect case for transformers, right?

**Reply by mjashanks** [reply]
Yea - I had a go at this with handlebars, and couldn't work out a way to do it.

Yes, this could be done with transformers. Also, you will be able to write javascript instead of handlebars, in the binding drawer. So, you could do it this way also. If you need it inside a table component, you'd need to use transformers though.

All of that is coming at the end of October.

**Reply by tobehn** [reply]
Thanks @mjashanks i am really looking forward to this step.

**Comment by tobehn** [comment, marked as answer]
In case anybody is trying to do something like this...

With the October release i was able to do this via the binding drawer in the actions of a save button so i can generate nice strings of (in my case) part numbers.
![image](https://user-images.githubusercontent.com/40893107/142248081-8042dc44-7ce8-449b-89cd-5665c14c810d.png)

```
const topLevel = $("RProducts.00_PBS-Products.PBS Top Level");
const assemSign = $("RAssemblies.01_PBS-Assemblies.Sign");
const num = $("Form.Fields.Number");
const numStr = num.toString();

let pbsName = topLevel + "-" + assemSign + "-" + numStr.padStart(4, '0');

return (pbsName)
```
this will then be written into the defined cell of name PBS-Number in the set table. W3-100-**0001** --> hooray :tada:

**Reply by mjashanks** [reply]
Brilliant - thanks for sharing!
